src/context/config.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {
+  createContext,
+  useContext,
+  FunctionComponent,
+  useState,
+  useEffect
+} from "react";
+import { AsyncStorage } from "react-native";
+import { LoadingView } from "../components/Loading";
+
+export enum AppMode {
+  production = "production",
+  staging = "staging"
+}
+
+interface Config {
+  appMode: AppMode;
+}
+
+interface ConfigContext {
+  config: Config; // context consumers should never get an undefined config
+  setConfigValue: <K extends keyof Config>(key: K, value: Config[K]) => void;
+}
+
+const CONFIG_KEY = "CONFIG";
+const DEFAULT_CONFIG: Config = { appMode: AppMode.production };
+
+const ConfigContext = createContext<ConfigContext>({
+  config: DEFAULT_CONFIG,
+  setConfigValue: () => {} // eslint-disable-line @typescript-eslint/no-empty-function
+});
+
+export const useConfigContext = (): ConfigContext =>
+  useContext<ConfigContext>(ConfigContext);
+
+export const ConfigContextProvider: FunctionComponent = ({ children }) => {
+  const [config, setConfig] = useState();
+
+  const setConfigValue: ConfigContext["setConfigValue"] = (key, value) => {
+    const nextConfig = {
+      ...config,
+      [key]: value
+    };
+    setConfig(nextConfig);
+    AsyncStorage.setItem(CONFIG_KEY, JSON.stringify(nextConfig));
+  };
+
+  const loadConfigFromStore = async (): Promise<void> => {
+    const configStr = await AsyncStorage.getItem(CONFIG_KEY);
+    if (configStr) {
+      setConfig(JSON.parse(configStr));
```

Comment:
there is one problem with this, if you add a new settings and there already exist a stored config, then there will be no default value provided for the new setting

=> https://github.com/Open-Attestation/identity-wallet/pull/63/files#diff-5689e8dfaa1efe9470061c2814ecf3e0R37

Reply:
Ok will add this in